src/utils/schema-v4-generator.js

```diff
@@ -0,0 +1,177 @@
+const type = require('is-type');
+const getType = require('./get-type');
+
+const SCHEMA_VERSION = 'http://json-schema.org/draft-04/schema#';
+
+class SchemaV4Properties {
+  constructor({ keysStrict, valuesStrict, typesStrict }) {
+    this.set({ keysStrict, valuesStrict, typesStrict });
+  }
+
+  set({ keysStrict, valuesStrict, typesStrict }) {
+    this.keysStrict = keysStrict || false;
+    this.valuesStrict = valuesStrict || false;
+    this.typesStrict = typesStrict || false;
+    return this;
+  }
+}
+
+class SchemaV4Generator {
+  constructor({ json, properties }) {
+    this.json = type.string(json) ? JSON.parse(json) : json;
+
+    this.schema = undefined;
+    this.properties = properties || new SchemaV4Properties({});
+
+    if (Array.isArray(this.properties.valuesStrict)) {
+      this.properties.valuesStrict = this.properties.valuesStrict.map(
+        (value) => {
+          return value.toLowerCase();
+        }
+      );
+    }
+  }
+
+  generate() {
+    this.schema = this.getSchemaForObject({
+      baseObject: this.json,
+      objectId: undefined,
+      firstLevel: true,
+      properties: this.properties
+    });
+
+    return this.schema;
+  }
+
+  getSchemaTypeFor(val) {
+    const _type = getType(val);
+
+    if (val == null) {
```

Comment:
I would use already introduced `isset(val)` to handle `undefined` as well.

Reply:
As of now, it would resolve to `'null` upon `null`, and to `undefined` when incase of `undefined`. It's hard to evaluate whether it's expected as a part of library's handling of that type.